AABB_tree/include/CGAL/AABB_tree.h

```diff
@@ -646,6 +646,22 @@ namespace CGAL {
 			else
 				return this->any_reference_point_and_id();
 		}
+		
+		//! Returns the datum (geometric object) represented `p`. 
+		//! The return type is 
+		//! - typename `Primitive::Datum_reference` if `Primitive` defines that type.
+		//! - typename `Primitive::Datum` if it doesn't.
```

Comment:
Both `AABBPrimitive` and `AABBPrimitiveWithSharedData` document that `datum()` returns `Data_reference`. Why do you have something different here?

Reply:
In the helper it tests the existance of Datum reference, saying that it is doing so for compatibility reasons. I didn't see any reason to hide the mechanism here.

Reply:
This was done for backward compatibility reasons, you can document that it returns `Datum_reference`, the helper class will handle compatibility issues internally.

Reply:
Ok.